LAVA_setup_guide.md

```diff
@@ -0,0 +1,172 @@
+# LAVA setup
+
+This document lists all the steps needed to setup a lava server and a worker.
+
+`These Lines` can be copy pasted to the command line. Elements in `<brackets>` need to be replaced with real values.
+
+## Building LAVA server and client
+
+Lava server and worker can be made from scratch or you can use the docker compose and RPi SD card image (recommended).
+The process below describes how to do it from scratch.
+
+### OS setup
+
+Lava uses debian buster as base for both server and workers.
+
+- Install debian buster https://www.debian.org/releases/buster/debian-installer/
+- Log in as root
+- `usermod -a -G sudo <username>`
+- `echo "deb https://apt.lavasoftware.org/release buster main" >> /etc/apt/sources.list`
+- you may now login as user
+- `wget https://apt.lavasoftware.org/lavasoftware.key.asc`
+- `sudo apt-key add lavasoftware.key.asc`
+- `sudo apt-get update`
+
+Note: although it is not officially supported, lava-server can be installed in Ubuntu (e.g. on WSL) by adding the below step:
+`sudo apt-get install postgresql && sudo service postgresql start`
+
+This resolves a dpkg-configure error that occurs because Postgres is not listening to port 5432 during installation
+
+### Server setup
+
+Here is the process you'll need to build the server from scratch. It's recommended you use the docker compose instead -
+see below. 
+
+#### Install LAVA server
+
+- `sudo apt install lava-server`
+- `sudo a2dissite 000-default`
+- `sudo a2enmod proxy`
+- `sudo a2enmod proxy_http`
+- `sudo a2ensite lava-server.conf`
+- `sudo service apache2 restart`
+
+#### Setup user
+
+- `sudo lava-server manage users add <usernane> --passwd <password>`
+- `sudo lava-server manage authorize_superuser --username <usernane>`
+
+#### Set website URL (this affect internal web interface links)
+
+- go to http://localhost/admin/sites/site/
+- click on example.com and delete it
+- add new one with the correct url
+- edit /usr/lib/python3/dist-packages/lava_server/settings/common.py
+    * search for SITE_ID and change it to SITE_ID = 2
+    * `sudo service lava-server-gunicorn restart`
+    * any time you change site domain you have to change the index to match
+
+#### Add Worker
+
+- `sudo lava-server manage workers add <WORKER_NAME>`
+    * don't do that on a single machine install, it will break lava
+    * set the hostname to the worker name
+- copy the token string to be used when configuring the worker
+
+#### Add device definition
+
+- create a new file in /etc/lava-server/dispatcher-config/devices with the name of the device (you can use the one from https://github.com/paul-szczepanek-arm/lava-prototype)
+    * You can just copy the whole directory https://github.com/paul-szczepanek-arm/lava-prototype/tree/master/lava-server to /etc/, otherwise follow steps below
```

Comment:
Why does it point to your local repository ? 
When a user should make a PR to add its device to the database ?

Reply:
because when I wrote this there wasn't an official one, thanks.

If the user wants to add a device he can make a new file and add it to his server. This repository is for us not for users. They can find their own way to manage devices. To be honest not sure what you're asking.

Reply:
I guess you're asking how we add devices: we add the type to this repo and rebuild the docker compose

Reply:
let me add that